pepys_import/resolvers/command_line_resolver.py

```diff
@@ -107,56 +121,148 @@ def resolve_sensor(self, data_store, sensor_name, sensor_type, host_id, privacy,
             f"Search for existing sensor on platform '{host_platform.name}'",
             f"Add a new sensor",
         ]
+        objects = (
+            data_store.session.query(data_store.db_classes.Sensor)
+            .filter(data_store.db_classes.Sensor.host == host_id)
+            .all()
+        )
+        objects_dict = {obj.name: obj for obj in objects}
+        if len(objects_dict) <= 7:
+            options.extend(objects_dict)
         if sensor_name:
             options[1] += f", default name '{sensor_name}'"
             prompt = f"Sensor '{sensor_name}' on platform '{host_platform.name}' not found. Do you wish to: "
         else:
             prompt = f"Sensor on platform '{host_platform.name}' not found. Do you wish to: "
-        choice = create_menu(prompt, options, validate_method=is_valid,)
 
-        if choice == str(1):
+        def is_valid_dynamic(option):
+            return option in [str(i) for i in range(1, len(options) + 1)] or option == "."
+
+        choice = create_menu(prompt, options, validate_method=is_valid_dynamic)
+        if choice == ".":
+            print("Quitting")
+            sys.exit(1)
+        elif choice == str(1):
             return self.fuzzy_search_sensor(
                 data_store, sensor_name, sensor_type, host_platform.platform_id, privacy, change_id
             )
         elif choice == str(2):
             return self.add_to_sensors(
                 data_store, sensor_name, sensor_type, host_id, privacy, change_id
             )
-        elif choice == ".":
-            print("Quitting")
-            sys.exit(1)
+        elif 3 <= int(choice) <= len(options):
+            selected_object = objects_dict[options[int(choice) - 1]]
+            if selected_object:
+                return selected_object
 
-    def resolve_privacy(self, data_store, change_id, data_type):
-        # Choose Privacy
-        privacy_names = [
-            "Search an existing classification",
-            "Add a new classification",
-        ]
-        choice = create_menu(
-            f"Ok, please provide classification for new {data_type}: ",
-            privacy_names,
-            validate_method=is_valid,
+    def resolve_reference(
```

Comment:
Could you add a docstring to this method, explaining the parameters? It's a little unclear at first look what text_name and field_name and so on do.

Reply:
Added:
https://github.com/debrief/pepys-import/pull/408/commits/887fdf8132ae3643f9be5fe7892f2fba5b66a097